state/address.go

```diff
@@ -221,21 +221,71 @@ func (st *State) APIHostPortsForClients() ([][]network.HostPort, error) {
 
 // APIHostPortsForAgents returns the collection of API addresses that should
 // be used by agents.
+// If the controller model is CAAS type, the return will be the controller
+// k8s service addresses in cloud service.
 // If there is no management network space configured for the controller,
 // or if the space is misconfigured, the return will be the same as
 // APIHostPortsForClients.
 // Otherwise the returned addresses will correspond with the management net space.
 // If there is no document at all, we simply fall back to APIHostPortsForClients.
 func (st *State) APIHostPortsForAgents() ([][]network.HostPort, error) {
-	hp, err := st.apiHostPortsForKey(apiHostPortsForAgentsKey)
+	if hps, err := st.apiHostPortsForOperator(); err == nil {
```

Comment:
We are ignoring potential fatal errors.
We need instead to get isCAAS from the model and call the correct method

Reply:
ic